app/assets/javascripts/dispatchers/frontDispatcher.js

```diff
@@ -7,7 +7,7 @@
     // chosen by a manager. In order to know which JS router/view to instantiate, we read
     // a global variable called "route" created inside the rails view matching each template
 
-    var routerName = 'Front' + window.route;
+    var routerName = 'Front' + (window.route || 'Homepage'); // Default to homepage.
```

Comment:
This is so the language selector will always render. 

homepage is basically the index route in our case.